lib/gringotts/gateways/mercadopago.ex

```diff
@@ -0,0 +1,368 @@
+  | `order_id`               | Order id issued by the merchant. Type- integer                                          |
+  | `customer_id`            | Unique customer id issued by the gateway. For new customer it must skipped. Type- string|
+  | `order_type`             | `"mercadopago"` or `"mercadolibre"` as per the order. Type- string                      |
+  | `installments`           | No of installments for payment. Type- integer                                           |
+
+  ## Registering your mercadopago account at `Gringotts`
+
+  After [making an account on mercadopago][credentials], head to the credentials and find
+  your account "secrets" in the `Checkout Transparent`.
+
+  | Config parameter | MERCADOPAGO secret |
+  | -------          | ----               |
+  | `:access_token`  | **Access Token**   |
+  | `:public_key`    | **Public Key**     |
+
+  > Your Application config **must include the `[:public_key, :access_token]` field(s)** and would look
+  > something like this:
+  >
+  >     config :gringotts, Gringotts.Gateways.Mercadopago,
+  >         public_key: "your_secret_public_key"
+  >         access_token: "your_secret_access_token"
+
+  [credentials]: https://www.mercadopago.com/mlb/account/credentials?type=basic
+
+  ## Note
+
+  * mercadopago processes money in the subdivided units (like `cents` in case of
+    US Dollar).
+  * Also, there is no way to set the currency of the transaction via the API. It
+    is automatically set from the merchant's account. Hence, if you've
+    configured your mercadopago account to work in Chilean Peso (`CLP`), make
+    sure that the `amount` argument is always a `Money.t` struct with the `:CLP`
+    as currency.
+
+  ## Supported currencies and countries
+
+  mercadopago supports the currencies listed [here][currencies].
+
+  [currencies]: https://api.mercadopago.com/currencies
+
+  ## Following the examples
+
+  1. First, set up a sample application and configure it to work with MERCADOPAGO.
+  - You could do that from scratch by following our [Getting Started][gs] guide.
+      - To save you time, we recommend [cloning our example
+      repo][example] that gives you a pre-configured sample app ready-to-go.
+          + You could use the same config or update it the with your "secrets"
+          as described [above](#module-registering-your-mercadopago-account-at-gringotts).
+
+  [gs]: https://github.com/aviabird/gringotts/wiki/
+  [home]: https://www.mercadopago.com
+  [example]: https://github.com/aviabird/gringotts_payment
+  """
+
+  @base_url "https://api.mercadopago.com"
+  use Gringotts.Gateways.Base
+  alias Gringotts.CreditCard
+  # The Adapter module provides the `validate_config/1`
+  # Add the keys that must be present in the Application config in the
+  # `required_config` list
+  use Gringotts.Adapter, required_config: [:public_key, :access_token]
+
+  alias Gringotts.{CreditCard, Money, Response}
+
```

Comment:
@mandarvaze `cvc_code_translator` and `avs_code_translator` find and add mappings.